src/core/tests/node_input_output.cpp

```diff
@@ -150,3 +155,307 @@ TEST(node_input_output, create_wrong_input_output) {
+    auto relu = make_shared<op::v0::Relu>(add1);
+    auto add2 = make_shared<op::v1::Add>(relu, param);
+
+    // add2 uses relu which uses add1
+    // Now if we try to make add1 use add2's output (creating a cycle in theory)
+    // This test ensures replace() handles it gracefully
+
+    ASSERT_EQ(add1->output(0).get_target_inputs().size(), 1);  // relu
+    ASSERT_EQ(add2->output(0).get_target_inputs().size(), 0);  // no consumers yet
+
+    // Add a consumer to add2 that is add1 (simulating the problematic case)
+    auto mul = make_shared<op::v1::Multiply>(add2, add1);
+
+    // Now replace add1's output with add2's output
+    add1->output(0).replace(add2->output(0));
+
+    // Check results
+    auto add2_targets = add2->output(0).get_target_inputs();
+    // Note: mul has 2 inputs from add2 (first operand) and add1 (second operand)
+    // After replace, mul's second input should also point to add2
+    EXPECT_EQ(add2_targets.size(), 3) << "add2 should have relu + mul(2 inputs) as targets";
+
+    set<Node*> target_nodes;
+    for (const auto& input : add2_targets) {
+        target_nodes.insert(input.get_node());
+    }
+
+    EXPECT_TRUE(target_nodes.count(relu.get()) > 0) << "relu should connect to add2";
+    EXPECT_TRUE(target_nodes.count(mul.get()) > 0) << "mul should connect to add2";
+    EXPECT_FALSE(target_nodes.count(add1.get()) > 0) << "add1 should not be in add2's targets";
+
+    EXPECT_EQ(add1->output(0).get_target_inputs().size(), 0) << "add1 should have no targets";
+}
+
+TEST(node_input_output, output_replace_empty_targets) {
+    // Corner case: replacing output that has no targets
+    auto param = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 3});
+    auto add1 = make_shared<op::v1::Add>(param, param);
+    auto add2 = make_shared<op::v1::Add>(param, param);
+    auto relu = make_shared<op::v0::Relu>(add2);
+
+    // add1 has no targets, add2 has relu
+    ASSERT_EQ(add1->output(0).get_target_inputs().size(), 0);
+    ASSERT_EQ(add2->output(0).get_target_inputs().size(), 1);
+
+    // Replace add1's output (no targets) with add2's output (has targets)
+    add1->output(0).replace(add2->output(0));
+
+    // Nothing should change since add1 had no targets
+    EXPECT_EQ(add1->output(0).get_target_inputs().size(), 0);
+    EXPECT_EQ(add2->output(0).get_target_inputs().size(), 1);
+    EXPECT_EQ(relu->input_value(0).get_node(), add2.get());
+}
+
+TEST(node_input_output, output_replace_with_parameter) {
+    // Corner case: replacing with parameter output
+    auto param = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 3});
+    auto add = make_shared<op::v1::Add>(param, param);
+    auto relu = make_shared<op::v0::Relu>(add);
+
+    // Initial state: param has 2 connections to add
+    ASSERT_EQ(param->output(0).get_target_inputs().size(), 2);
+
+    // Replace add's output with parameter's output
+    add->output(0).replace(param->output(0));
```

Comment:
Here there is no cycling connection model is changed to 

Param -> Add 
          \-> Relu

It is still valid connection.

The change of `replace` shouldn't be modified it looks there is a need for some kind transformation util which replace output and remove this node from other node inputs.

The param is still connected to add Node.
```suggestion
    add->output(0).replace(param->output(0));
    add->set_arguments(ov::OutputVector{});
```
It will give same effect

Reply:
I fixed unit tests and revert replace()